mbbsd/mbbsd.c

```diff
@@ -1192,6 +1194,22 @@ user_login(void)
     setproctitle("%s: %s", margs, cuser.userid);
 #endif
 
+    // ensuring the contact email if lastlogin before sometime.
+#ifdef USEREC_EMAIL_IS_CONTACT
+    #ifdef ENSURE_CONTACT_EMAIL_LASTLOGIN
+    if(cuser.lastlogin < ENSURE_CONTACT_EMAIL_LASTLOGIN) {
```

Comment:
No. The lastlogin check should be a separate commit.

There should be two separate commits:
a) Force user to add their contact email if the current one is invalid.
b) Force user to update their contact email if their `lastlogin` is before some timestamp.
The second item is optional and should be guarded with `#ifdef FORCE_UPDATE_CONTACT_EMAIL_LASTLOGIN`

Reply:
Basically, you should add the force update stuff after [4b89509](https://github.com/ptt/pttbbs/commit/4b89509731a7adafab84d4d7465eae885b35867f), but with the style in that commit fixed as I mentioned before.

Reply:
Space between `if`/`while` and `(` please.

Reply:
> Basically, you should add the force update stuff after [4b89509](https://github.com/ptt/pttbbs/commit/4b89509731a7adafab84d4d7465eae885b35867f), but with the style in that commit fixed as I mentioned before.

Sorry for the wrong place of the code.
`lastlogin` is updated in `pwcuLoginSave` and it is after `check_register`
I'll change the code block accordingly.